util/promlint/promlint.go

```diff
@@ -232,17 +295,43 @@ func metricUnits(m string) (unit string, base string, ok bool) {
 // Units and their possible prefixes recognized by this library.  More can be
 // added over time as needed.
 var (
-	baseUnits = []string{
-		"amperes",
-		"bytes",
-		"candela",
-		"grams",
-		"kelvin", // Both plural and non-plural form allowed.
-		"kelvins",
-		"meters", // Both American and international spelling permitted.
-		"metres",
-		"moles",
-		"seconds",
+	// map a unit to the appropriate base unit.
+	units = map[string]string{
+		// Base units.
+		"amperes": "amperes",
+		"bytes":   "bytes",
+		"candela": "candela",
+		"celsius": "celsius", // Celsius is more common in practice than Kelvin.
+		"grams":   "grams",
+		"joules":  "joules",
+		"meters":  "meters", // Both American and international spelling permitted.
+		"metres":  "metres",
+		"moles":   "moles",
+		"seconds": "seconds",
+		"volts":   "volts",
+
+		// Non base units.
+		// Time.
+		"minutes": "seconds",
+		"hours":   "seconds",
+		"days":    "seconds",
+		"weeks":   "seconds",
+		// Temperature.
+		"kelvin":     "celsius",
+		"kelvins":    "celsius",
+		"fahrenheit": "celsius",
+		"rankine":    "celsius",
+		// Length.
+		"inches": "meters",
+		"yards":  "meters",
+		"miles":  "meters",
+		// Bytes.
+		"bits": "bytes",
+		// Energy.
+		"calories": "joules",
+		// Weight.
```

Comment:
Technically weight is newtons :)

Reply:
My physics professors probably all just disowned me :smile:.

Shall I also make the docs technically correct?